examples/unreliable-datagram/README.md

```diff
@@ -0,0 +1,42 @@
+# Unreliable Datagram
+
+This folder contains an example of sending and receiving an 'unreliable datagram' in `s2n-quic`. An 'unreliable datagram' is data that is not retransmitted upon packet loss, and is defined in https://datatracker.ietf.org/doc/rfc9221. This is in contrast to 'stream data' which is retransmitted during packet loss. Stream data is the preferred method of sending application data, as it comes with support for features like flow control and packet retransmission. Additionally, unreliable datagram users are responsible for making sure their datagrams fit into the packet space given, as packet fragmentation/assembly is not allowed. A common use-case for unreliable datagrams is video or audio streaming, where individual datagrams are small and some loss is acceptable.
+
+# Default Datagram Implementation
+
+`s2n-quic` contains a default implementation of the unreliable datagram sender and receiver. This implementation is minimal, given the many different ways that unreliable datagrams could be used. Users should look out for the fact that datagrams that are too large to be sent will be automatically dropped, and old datagrams will be dropped in favor of new datagrams from the peer if there is no more space on the receiving queue.
```

Comment:
> Users should look out for the fact that datagrams that are too large to be sent will be automatically dropped, and old datagrams will be dropped in favor of new datagrams from the peer if there is no more space on the receiving queue.

Are there other caveats/assumptions (like stream and datagram prioritization) and are they documented somewhere? Some of this might also belong in the general rust doc.

Reply:
We also have a [docs](https://github.com/aws/s2n-quic/tree/main/docs) folder.. but not sure if thats been updated since public launch.

Reply:
There are other caveats, they are documented in code comments and "comments above functions". I thought those translated to the rustdoc stuff, but I could be wrong.

Reply:
Makes sense. Thanks for adding more docs!